### DISCUSSION TITLE
Sidekiq: Mastodon::UnexpectedResponseError: returned code 500

### DISCUSSION THREAD

**Discussion by felixriehm** [mastodon/mastodon#22631] [Q&A]
Im hosting a mastodon instance on Ubuntu 20.04.5 LTS without Docker. When i try to reply to a toot from another instance i get the following sidekiq warning: 

```
pid=78022 tid=1767a WARN: {"context":"Job raised exception","job":{"retry":16,"queue":"push","dead":false,"args":["{\"@context\":\"https://www.w3.org/ns/activitystreams\",\"id\":\"https://social.mpdl.mpg.de/users/riehm/statuses/109551684372195833/activity\",\"type\":\"Announce\",\"actor\":\"https://social.mpdl.mpg.de/users/riehm\",\"published\":\"2022-12-21T12:34:03Z\",\"to\":[\"https://www.w3.org/ns/activitystreams#Public\"],\"cc\":[\"https://c.im/users/Glastomichelle\",\"https://social.mpdl.mpg.de/users/riehm/followers\"],\"object\":\"https://c.im/users/Glastomichelle/statuses/109551008608445365\"}",109312717925186172,"https://c.im/users/Glastomichelle/inbox"],"class":"ActivityPub::DeliveryWorker","jid":"c6f4406e2ab99811ab4406ce","created_at":1671626043.2972803,"enqueued_at":1671651888.1047504,"error_message":"https://c.im/users/Glastomichelle/inbox returned code 500","error_class":"Mastodon::UnexpectedResponseError","failed_at":1671626044.3160274,"retry_count":10,"retried_at":1671641768.5740547}}
pid=78022 tid=1767a WARN: Mastodon::UnexpectedResponseError: https://c.im/users/Glastomichelle/inbox returned code 500
pid=78022 tid=1767a WARN: /home/mastodon/live/app/workers/activitypub/delivery_worker.rb:54:in `block (3 levels) in perform_request'
```

When i check the toot on the original server i dont see the reply. I have checked this with an account on sueden.social and did the same action. With that account i can see the reply on the toot on the original server.

I also checked my domain with ssllabs com. The report shows me a A+ without any issues ([mentioned here](https://github.com/mastodon/mastodon/issues/6449)). I also checked following ([mentioned here](https://github.com/mastodon/mastodon/issues/8375))
`curl -v http://social.mpdl.mpg.de/.well-known/host-meta `-> no problems

and `curl "https://social.mpdl.mpg.de/.well-known/webfinger?resource=riehm@social.mpdl.mpg.de" `-> no problems

If i try `dpkg -l | grep -E 'openssl|libssl|ca-certificates'` ([mentioned here](https://github.com/mastodon/mastodon/discussions/17857)) i get :

```
ii  ca-certificates                       20211016ubuntu0.20.04.1           all          Common CA certificates
ii  libssl-dev:amd64                      1.1.1f-1ubuntu2.16                amd64        Secure Sockets Layer toolkit - development files
ii  libssl1.1:amd64                       1.1.1f-1ubuntu2.16                amd64        Secure Sockets Layer toolkit - shared libraries
ii  libxmlsec1-openssl:amd64              1.2.28-2                          amd64        Openssl engine for the XML security library
ii  openssl                               1.1.1f-1ubuntu2.16                amd64        Secure Sockets Layer toolkit - cryptographic utility
ii  python3-openssl                       19.0.0-1build1                    all          Python 3 wrapper around the OpenSSL library
```

Searching for other users mostly work. Also following works fine. However, when searching for a user, there are some users like "@JetForMe@geekstodon.com" that return "503 Remote data could not be fetched" on the web UI and status 503 in the journalctl log of mastodon-web `method=GET path=/api/v2/search format=html controller=Api::V2::SearchController action=index status=503 duration=184.10 view=0.57 db=1.26` (works fine with my sueden.social account).

How can i fix these problems? Is there any further information i can provide you?

**Comment by lazzarello** [comment]
I was getting 503 Service Unavailable errors on follow requests from my own server but only to mastoson.social URLs. They are gone after three hours. I suspect the service was actually unavailable.

**Comment by vengefulpunk** [comment]
@felixriehm Did you ever resolve this issue? I am running into it now. 

The instances I am attempting to communicate with are online(it is 3 of them doing it as well)

**Comment by felixriehm** [comment]
@vengefulpunk no, after a few days the reply suddenly appeared.
